tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -933,6 +933,37 @@ func (az *AzureProvider) destroyDisk(ctx context.Context, vmInfo compute.Virtual
 	return
 }
 
+// DestroyWindowsVMWithResources destroys the Windows VMs along with all the errors associated with it
+func (az *AzureProvider) DestroyWindowsVM(vmName string) error {
+	ctx := context.Background()
+	_, nicName := az.getNICname(ctx, vmName)
+	_, ipName := az.getIPname(ctx, vmName)
+	vmInfo, _ := az.vmClient.Get(ctx, az.resourceGroupName, vmName, compute.InstanceView)
+
+	log.Printf("deleting the resources associated with instance %s", vmName)
```

Comment:
nit: I'm not a fan of this pattern of logging the current action within functions like this. If we want to log this I would rather the caller of `DestroyWindowsVM` log it.

Reply:
I don't like it either, this is a problem of the way the code is structured. I've moved it to the caller.